### PR TITLE
[LLVM] Support LLVM 10 on macOS

### PR DESCRIPTION
**Pull request by TH3CHARLie**
Related issue = #655 (if any)

- [[Click here for the format server]](http://kun.csail.mit.edu:31415/)
- [[Click here for coverage report]](http://codecov.io/gh/taichi-dev/taichi/)

This PR fix name mangling in JIT CPU backend to support LLVM 10 on macOS.

### CODE REVIEW

**Comment by TH3CHARLie** [comment]
@k-ye Would you mind testing this PR locally? I double-check it but it would be great if you can verify it on your macOS machine. Especially on the `#ifdef` part, feel free to remove it and test the  `ES->intern` on macOS, I observe that it would crash on macOS due to incorrect name mangling.

The LLVM 10 source I built against is https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/llvm-10.0.0.src.tar.xz

File: taichi/backends/cpu/jit_cpu.cpp [hunk 1]
```diff
@@ -129,15 +129,23 @@ class JITSessionCPU : public JITSession {
 
   void *lookup(const std::string Name) override {
     std::lock_guard<std::mutex> _(mut);
+#ifdef __APPLE__
+    auto symbol = ES.lookup(all_libs, Mangle(Name));
```
Comment (rexwangcc) [inline taichi/backends/cpu/jit_cpu.cpp:133]: Just a curious question, I'm not familiar with what type of mangling LLVM will do on mac for IR, it's it just some sort of `_` or `__`?

Comment (TH3CHARLie) [reply taichi/backends/cpu/jit_cpu.cpp:133]: My experiment shows the mangling results of function `foo` will be `_foo{i}`(i will start from 0)

**Comment by k-ye** [comment]
Is there an ETA for when this can get merged... That helps me decide whether I should re-install LLVM 8.0.1...

**Comment by yuanming-hu** [comment]
Given everyone has approved, I'll go merge this PR. Free free to continue the discussions in #655. Thank you guys for helping with this!

**Comment by TH3CHARLie** [comment]
> LGTM! Thanks. Maybe we should use `Mangle(Name)` instead of `ES.intern(Name)` for all platforms. I'll confirm if that works in a future PR.

That's what I am also curious about because I see you've made that change in #1200 from `Mangle` to `ES.intern` on windows platform. Not sure if they have the same behavior, especially when different platforms are involved.

**Comment by yuanming-hu** [comment]
> That's what I am also curious about because I see you've made that change in #1200 from `Mangle` to `ES.intern` on windows platform. Not sure if they have the same behavior, especially when different platforms are involved.

Yeah. I had to resolve a mysterious bug on the JIT part on Windows. Clearly LLVM was not so well documented there, so I could only tweak a lot of places to make things work. And once it works, I didn't do further attempt roll my tweaks back...

**Pull request merged**